subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/configurations/model/ConfigurationReportModelFactory.java

```diff
@@ -0,0 +1,196 @@
+            type = ReportConfiguration.Type.LEGACY;
+        } else if (configuration.isCanBeResolved()) {
+            type = ReportConfiguration.Type.RESOLVABLE;
+        } else if (configuration.isCanBeConsumed()) {
+            type = ReportConfiguration.Type.CONSUMABLE;
+        } else {
+            type = null;
+        }
+
+        return new ReportConfiguration(configuration.getName(), configuration.getDescription(), type, attributes, capabilities, artifacts, variants, extendedConfigurations);
+    }
+
+    private ReportArtifact convertPublishArtifact(PublishArtifact publishArtifact, FileResolver fileResolver) {
+        return new ReportArtifact(publishArtifact.getName(), fileResolver.resolveForDisplay(publishArtifact.getFile()), publishArtifact.getClassifier(), publishArtifact.getType());
+    }
+
+    private ReportSecondaryVariant convertConfigurationVariant(ConfigurationVariant variant, FileResolver fileResolver) {
+        final List<ReportAttribute> attributes = Collections.unmodifiableList(variant.getAttributes().keySet().stream()
+            .map(a -> convertAttributeInContainer(a, variant.getAttributes()))
+            .sorted(Comparator.comparing(ReportAttribute::getName))
+            .collect(Collectors.toList()));
+        final List<ReportArtifact> artifacts = Collections.unmodifiableList(variant.getArtifacts().stream()
+            .map(publishArtifact -> convertPublishArtifact(publishArtifact, fileResolver))
+            .sorted(Comparator.comparing(ReportArtifact::getName))
+            .collect(Collectors.toList()));
+
+        return new ReportSecondaryVariant(variant.getName(), variant.getDescription().orElse(null), attributes, artifacts);
+    }
+
+    private ReportAttribute convertAttributeInContainer(Attribute<?> attribute, AttributeContainer container) {
+        @SuppressWarnings("unchecked") Attribute<Object> key = (Attribute<Object>) attribute;
+        Object value = container.getAttribute(key);
+        return new ReportAttribute(key, value);
+    }
+
+    private ReportAttribute convertUncontainedAttribute(Attribute<?> attribute) {
+        @SuppressWarnings("unchecked") Attribute<Object> key = (Attribute<Object>) attribute;
+        return new ReportAttribute(key, null);
+    }
+
+    private ReportCapability convertCapability(Capability capability) {
+        return new ReportCapability(capability.getGroup(), capability.getName(), capability.getVersion(), false);
+    }
+
+    private ReportCapability convertDefaultCapability(Project project) {
+        return new ReportCapability(Objects.toString(project.getGroup()), project.getName(), Objects.toString(project.getVersion()), true);
+    }
+
+    /**
+     * This class can examine a project to deterimne which {@link Attribute}s in its schema have {@link org.gradle.api.internal.attributes.CompatibilityRule CompatibilityRule}s or
+     * {@link org.gradle.api.internal.attributes.DisambiguationRule DisambiguationRule}s defined.
```

Comment:
```suggestion
     * This class can examine a project to deterimne which {@link Attribute}s in its schema have {@link CompatibilityRule}s or
     * {@link DisambiguationRule}s defined.
```
Might need to add imports for this, but I think it's much better to do that and have these be shorter.

Reply:
So because I added the simple name again after the FQN, they will appear as just the simple name in the rendered javadoc.  If the only reference to a class in a source file is as a javadoc `@link`, it's better to do this rather than link just the simple name and force the addition of an import, which might get read by other tooling and miscommunicates to the reader a usage which is not _really_ present in the source.

Reply:
Hmm, fair, I am just used to reading Javadoc in its original source form, where the shorter text increases readability.